d2lbook/translate.py

```diff
@@ -29,6 +29,8 @@ def __init__(self, cf: config.Config, commit: str):
         if os.path.exists(self.repo_dir):
             self.repo = git.Repo(self.repo_dir)
             logging.info(f'Pulling from {self.url} into {self.repo_dir}')
+            # reset to origin/master before pulling updates
```

Comment:
Capitalize the first letter in code comments

Reply:
Why do we need to reset to origin/master?
For example, users may run
d2lbook translate <filename>  # This asks for origin/master
d2lbook translate --commit 8c24d68 <filename>  # This asks for commit 8c24d68, not origin/master

Reply:
yes, but in the next line, we pull changes from origin/master to keep the repo updated. Now the error caused earlier was due to this pull. Here's how -> after we reset the repo head to a particular commit in our previous run, the next time we try to pull from origin/master, we run into a merge conflict that needs to be resolved manually.

Hence the best way is to always reset to origin/master ([line 33](https://github.com/d2l-ai/d2l-book/blob/8a56267645c8893dc81215561620e22cefd85b2e/d2lbook/translate.py#L33)), then pull the changes ([line 34](https://github.com/d2l-ai/d2l-book/blob/8a56267645c8893dc81215561620e22cefd85b2e/d2lbook/translate.py#L34)) and update the repo to the latest and then simply move the head to that specific commit ([line 41](https://github.com/d2l-ai/d2l-book/blob/8a56267645c8893dc81215561620e22cefd85b2e/d2lbook/translate.py#L41)).

Reply:
> Capitalize the first letter in code comments

Is there a fixed styled guide, I just followed the approach in some other comments to maintain the style which is caps off with the comments. Anyway, I've updated the comment.

Reply:
Thanks. I am curious about the semantics of [active branch](https://gitpython.readthedocs.io/en/stable/reference.html?highlight=active_branch#git.repo.base.Repo.active_branch?)

Suppose that a repo has two branches: master and r0.17, where the HEAD commit on r0.17 is 0123abcd. 

When users run:
`d2lbook translate --commit 0123abcd <filename>`

Does the active branch point to r0.17 not master? If yes, when users run the above command again, [line 33](https://github.com/d2l-ai/d2l-book/pull/40/files#diff-f4341051f8e7e3d3916e19a41f829009baf39736a303884f6b27734f080bdb78R33) will reset to origin/r0.17, not origin/master as indicated in your [comment](https://github.com/d2l-ai/d2l-book/pull/40/files#diff-f4341051f8e7e3d3916e19a41f829009baf39736a303884f6b27734f080bdb78R32) (do you need to revise your comment). 

Does [line 34](https://github.com/d2l-ai/d2l-book/pull/40/files#diff-f4341051f8e7e3d3916e19a41f829009baf39736a303884f6b27734f080bdb78R34) pull the corresponding upstream (origin/r0.17 in this case) to the active branch (r0.17 in this case) **only**?

Could you [double-check](https://gitpython.readthedocs.io/en/stable/tutorial.html?highlight=remotes#handling-remotes)? Thanks!

Reply:
>Does the active branch point to r0.17 not master? 

NO. active branch points to master in this case. But after running `d2lbook translate --commit 0123abcd <filename>` the repo has been reset to this commit `0123abcd` from [line 41](https://github.com/d2l-ai/d2l-book/blob/8b8c2a7d2490e40841a61d848e04737a88acd81e/d2lbook/translate.py#L41) which is why the master branch and `origin/master` have now diverged. When you pull again from origin/master ([line 34](https://github.com/d2l-ai/d2l-book/pull/40/files#diff-f4341051f8e7e3d3916e19a41f829009baf39736a303884f6b27734f080bdb78R34)) this results in merge conflict. It is not pulling from r0.17 as that is not the active branch. **Resetting to a particular commit doesn't mean that the branch has been switched.**